src/SAML2/Compat/ContainerSingleton.php

```diff
@@ -45,4 +46,25 @@ public static function initSspContainer(): Container
     {
         return new Container();
     }
+
+
+    /**
+     * @param string $className
+     * @param array $qualifiedName
+     * @return void
+     */
+    public static function registerClass(string $className, array $qualifiedName): void
+    {
+        $this->registry[$qualifiedName] = $className;
+    }
+
+
+    /**
+     * @param string $className
+     * @return array|false
+     */
+    public static function getRegisteredClass(string $className)
+    {
+        return array_search($className, $this->registry);
```

Comment:
```suggestion
    public static function getRegisteredClass(string $className)
    {
        return array_search($className, $this->registry);
```

What we need to search for is the namespace and element name. We need the registry to be able to run `fromXML()` on arbitrary classes we don't know about on beforehand. The only piece of information that we have when we need to call those classes is the namespace and the name of an elementt we want to process.